src/include/homestore/index/index_table.hpp

```diff
@@ -32,57 +33,51 @@ template < typename K, typename V >
 class IndexTable : public IndexTableBase, public Btree< K, V > {
 private:
     superblk< index_table_sb > m_sb;
+    shared< MetaIndexBuffer > m_sb_buffer;
 
 public:
     IndexTable(uuid_t uuid, uuid_t parent_uuid, uint32_t user_sb_size, const BtreeConfig& cfg) :
             Btree< K, V >{cfg}, m_sb{"index"} {
+        // Create a superblk for the index table and create MetaIndexBuffer corresponding to that
         m_sb.create(sizeof(index_table_sb));
         m_sb->uuid = uuid;
+        m_sb->ordinal = hs()->index_service().reserve_ordinal();
```

Comment:
We generate the ordinal for each index and store it in superblk, so upon recovery, we set the ordinal bit. Why ordinal? We need to serialize the btree/index each split/merge transaction record and to preserve the space (and reduce data written during CP), we use ordinal.